### PR TITLE
fix(typescript_indexer): add anon signature part for function expressions

### PR DESCRIPTION
**Pull request by ayazhafiz**
Function expressions like `(function())` have no name but introduce a
scope. Add an anonymous name for them when they are encountered in
generated a VName signature.

### CODE REVIEW

**Comment by ayazhafiz** [comment]
cc @evmar

**Comment by evmar** [comment]
I think you can add a test like

```
const x, y;
function() { const x; y;}
```

and checks that the xs are different and ys are same (maybe the latter isn't necessary).

**Comment by ayazhafiz** [comment]
> I think you can add a test like
> 
> ```
> const x, y;
> function() { const x; y;}
> ```
> 
> and checks that the xs are different and ys are same (maybe the latter isn't necessary).

done.

File: kythe/typescript/testdata/function.ts [hunk 1]
```diff
@@ -84,3 +86,14 @@ function bindingTest({aa, bb: {cc, dd}}, [ee, [ff]]: NestedArray) {
...
   dd = ff;
 }
+
+// Test function expressions introducing an anonymous scope
+//- @ev defines/binding Ev
+//- @an defines/binding An
+let ev, an;
+(function() {
+//- !{ @ev ref Ev }
+let ev;
```
Comment (evmar) [inline kythe/typescript/testdata/function.ts]: This is defines/binding, not ref, so I think it's always true even without your change.

Comment (ayazhafiz) [reply kythe/typescript/testdata/function.ts]: Fixed.

**Comment by ayazhafiz** [comment]
@evmar does this look okay?

**Comment by ayazhafiz** [comment]
@kythe PTAL

**Pull request merged**